File: NinetyNineBottles.swift
public class NinetyNineBottles {
    
    let beer:Int = 99
    
    func playSong() {
        
        for i in (1...beer).reversed() {
            
            let next = i == 1 ? 0 : (i-1)
            
            print("\(i) \(self.bottleString(beer: i)) of beer on the wall, \(i) \(self.bottleString(beer: i)) of beer.\n"
                + "Take one down and pass it around, "
                + "\(next) \(self.bottleString(beer: i-1))" + " of beer on the wall.")
            
        }
        
    }
    
    func bottleString(beer:Int) -> String {
        return beer == 1 ? "bottle" : "bottles"
    }
    
}

// Play 99 Bottles of Beer
let beer = NinetyNineBottles()
beer.playSong()

File: NinetyNineBottles.rb
def bottle_string(beer)
    beer == 1 ? "bottle" : "bottles"
end

def play_song
    puts "[Start song]"

    beer = 99

    while beer > 0
        next_beer = beer == 1 ? "no more" : beer - 1
        next_bottles = next_beer == "no more" ? bottle_string(0) : bottle_string(next_beer)

        puts "#{beer} #{bottle_string(beer)} of beer on the wall, #{beer} #{bottle_string(beer)} of beer."
        puts "Take one down and pass it around, #{next_beer} #{next_bottles} of beer on the wall.\n"
        
        beer -= 1
    end

    puts "No more bottles of beer on the wall, no more bottles of beer."
    puts "There's nothing else to fall, because there's no more bottles of beer on the wall."

    puts "[End Song]"
end

play_song
